tools/import_users/import_users.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf8 -*-
+
+import argparse
+import datetime
+import errno
+import logging
+import os
+import re
+import sys
+import yaml
+import json
+import csv
+import pydash
+from pydash import _
+from xlrd import open_workbook
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..'))
+from safetypy import safetypy as sp
+from tools.export_users import export_users
+# Possible values here are DEBUG, INFO, WARN, ERROR and CRITICAL
+LOG_LEVEL = logging.DEBUG
+actions = {}
+
+def process_desired_state(server_state, desired_state):
+    """
+    Processes the user provided input file and compares with the user data on the server
+    :param server_state: The list of all users and their associated groups in the server
+    :param desired_state: The input provided by the user
+    :return: None
+    """
+    group_id =[]
+    with open(desired_state) as csvDataFile:
+        csvReader = csv.reader(csvDataFile)
+        next(csvReader, None)
+        for email, lastname, firstname, groups in csvReader:
+            group_list = []
+            if groups != "":
+                group_list = groups.split(",")
+                group_list = [group.strip(' ') for group in group_list]
+            if email not in [user for user in server_state]:
+                actions[email] = {'action': 'add', 'groups': group_list, 'user_id':'', 'user_data': {'firstname': firstname, 'lastname': lastname, 'email': email} }
+            else:
+                group_names_server = server_state[email]['groups'][0::2]
+                group_names_desired = groups.split(',')
+                group_names_desired = [group.strip(' ') for group in group_names_desired]
+                group_diff = [i for i in group_names_desired if i not in group_names_server]
+                if group_diff != []:
+                    actions[email] = {'action': 'add to group', 'groups': group_diff, 'user_id': server_state[email]['user_id'] }
+
+
+def process_server_state(server_state, desired_state):
+    """
+    Processes the user data as per the server and compares with the input file provided by user
```

Comment:
Determines any actions that need to happen because the server state is different than the desired state